src/Adapter/Product/ListParametersUpdater.php

```diff
@@ -70,12 +70,42 @@ public function buildListParameters(
         array $persistedFilterParameters,
         array $defaultFilterParameters
     ) {
-        return [
-            'offset' => (int) $this->getParameter('offset', $queryFilterParameters, $persistedFilterParameters, $defaultFilterParameters),
-            'limit' => (int) $this->getParameter('limit', $queryFilterParameters, $persistedFilterParameters, $defaultFilterParameters),
-            'orderBy' => (string) $this->getParameter('orderBy', $queryFilterParameters, $persistedFilterParameters, $defaultFilterParameters),
-            'sortOrder' => (string) $this->getParameter('sortOrder', $queryFilterParameters, $persistedFilterParameters, $defaultFilterParameters),
+        $filters = [
+            'offset' => (int) $this->getParameter(
+                'offset',
+                $queryFilterParameters,
+                $persistedFilterParameters,
+                $defaultFilterParameters
+            ),
+            'limit' => (int) $this->getParameter(
+                'limit',
+                $queryFilterParameters,
+                $persistedFilterParameters,
+                $defaultFilterParameters
+            ),
+            'orderBy' => (string) $this->getParameter(
+                'orderBy',
+                $queryFilterParameters,
+                $persistedFilterParameters,
+                $defaultFilterParameters
+            ),
+            'sortOrder' => (string) $this->getParameter(
+                'sortOrder',
+                $queryFilterParameters,
+                $persistedFilterParameters,
+                $defaultFilterParameters
+            ),
         ];
+
+        /*
+         * We need to force the sort order when the order by
+         * is set to position_ordering
+         */
+        if ('position_ordering' === $filters['orderBy']) {
```

Comment:
ok but I think it should be done in `cleanFiltersForPositionOrdering`

Reply:
I tought about it but this function is called after the first usage of `$orderBy`.
ProductController::150 => getCatalogPorduct
and the clean is at 191 :(

Reply:
you're right, this controller is a nightmare..
I'm thinking of better ways to manage this, but none seems acceptable..

Reply:
We will have to change everything in the future, we need to refactor how the old migrated controllers have been done :/